projects/fw_103_hw/src/adc_converter.c

```diff
@@ -0,0 +1,61 @@
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+static void prv_button_press(const GpioAddress *address, void *context) {
+  uint16_t adc_data = 0;
+  GpioAddress *read_channel = (GpioAddress *)context;
+  adc_read_converted(*read_channel, &adc_data);
+  LOG_DEBUG("%d\n", adc_data);
+}
+
+int main(void) {
+  interrupt_init();
+  soft_timer_init();
+  gpio_init();
+  gpio_it_init();
+
+  GpioAddress read_addr = { .port = GPIO_PORT_A, .pin = 6 };
+  GpioAddress button_addr = { .port = GPIO_PORT_B, .pin = 2 };
+
+  GpioSettings read_settings = {
+    .direction = GPIO_DIR_IN,
+    .state = GPIO_STATE_LOW,
+    .resistor GPIO_RES_NONE,
+    .alt_function GPIO_ALTFN_ANALOG,
+  };
+
+  GpioSettings button_settings = {
+    .direction = GPIO_DIR_IN,
+    .state = GPIO_STATE_LOW,
+    .resistor GPIO_RES_NONE,
+    .alt_function GPIO_ALTFN_ANALOG,
```

Comment:
Same, this won't compile. (You could also consider using one gpio settings object since they're identical.)